File: src/Common/TreeMon.Models/Inventory/InventoryItem.cs
// Copyright (c) 2017 TreeMon.org.
//Licensed under CPAL 1.0,  See license.txt  or go to http://treemon.org/docs/license.txt  for full license details.
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace TreeMon.Models.Inventory
{
    [Table("Inventory")]
    public class InventoryItem: Item, INode
    {

        public InventoryItem():base()
        {
            this.UUIDType = "InventoryItem";
        }

        public string LocationUUID { get; set; }

        /// <summary>
        /// store,vehicle, room etc..
        /// </summary>
        public string LocationType { get; set; }

        public float Quantity { get; set; }

        public string ReferenceType { get; set; }   //  product, item, user, ballast, plant

        public string ReferenceUUID { get; set; } //id of the item in inventory if we have to break it down to individual items.


        //display in web store.
        public bool Published { get; set; }

        public DateTime ItemDate { get; set; }

        public string DateType { get; set; } //expires, end of cycle ....

        public string VendorUUID { get; set; }

        public string Link { get; set; }

        public string LinkProperties { get; set; }

        //TODO inventory log. show when items added, removed, etc. 
        //public DateTime DateAdded { get; set; }
        //public string AddedByUUID { get; set; }
    }
}

File: src/Client/TreeMon.Mobile/src/app/models/inventoryitem.ts
// Copyright 2015, 2017 TreeMon.org.
// Licensed under CPAL 1.0,  See license.txt  or go to http://treemon.org/docs/license.txt  for full license details.

import { Node } from './node';

export class InventoryItem extends Node {

    Condition: string;

    DetailView: boolean;

    Quantity: number;

    ReferenceType: string;

    ReferenceUUID: string;

    Virtual: boolean;

    LocationUUID: string;

    Published: boolean;

    CategoryUUID: string;

    GroupUUID: string;

    Custom: boolean;

    CategoryName: string;

    Discount: number;

    DiscountType: string;

    MarkUp: number;

    MarkUpType: string;

    Price: number;

    SKU: string;

    ManufacturerUUID: string;

    Expires: string;

    SerialNumber: string;

    SystemUUID: string;

    Weight: number;

    WeightUOM: string;

    UOM: string; // this is not in the server model.

    CartItemUUID: string;

    Link: string;

    LinkProperties: string;

    LocationType: string;

    Description: string;
}
